packages/material-ui-system/src/index.d.ts

```diff
@@ -202,5 +202,7 @@ export const typography: SimpleStyleFunction<
 >;
 export type TypographyProps = PropsFor<typeof typography>;
 
+export const visuallyHidden: CSSProperties;
```

Comment:
Adding `@material-ui/styles` seems wrong, but I needed the `CSSProperties` type, otherwise typescript was complaining when I was using the `visuallyHidden` in the `makeStyles`.

I tried using `CSS.Properties`, but ts was throwing an error:
![image](https://user-images.githubusercontent.com/4512430/84577737-4ec34400-adbf-11ea-9dbe-624f02805d9f.png)

Reply:
It sounds like the same problem than: #20379.

```suggestion
export const visuallyHidden: CSS.Properties;
```

Reply:
I did the changes, should I do a cast to `CSSProperties` in he TS example?

```diff
const useStyles = makeStyles((theme) => ({
-  visuallyHidden,
+  visuallyHidden: visuallyHidden as CSSProperties,
}));
```
I was expecting the build will fail, that's why I went initially with the `CSSProperties`

Reply:
Yeah, I think that it's the correct call. I think that we should optimize for styled components:

```jsx
import styled from 'styled-components';
import { visuallyHidden } from '@material-ui/system';
import { CSSProperties } from '@material-ui/styles';

const Span = styled('span')(visuallyHidden);

const useStyles = makeStyles((theme) => ({
  visuallyHidden: visuallyHidden as CSSProperties,
}));
```

Reply:
Or should we maybe use `styled-components` in the example?

Reply:
the `yarn:typescript:formatted` are conflicting with `yarn typescript` at this moment because of the case

Reply:
Yeah, it would make sense for the new demo, and only use CSSProperties for the Table demo.

I would be eager to fix the underlying issue: #20379. It seems to be an important one to address. The idea that we can't use the theme's helpers (breakpoints, typography, mixins) with styled-components and emotion sounds wrong. I think that we could try to find the intersection type between JSS, styled-components, and emotion. So that `visuallyHidden` can be used anywhere without any TypeScript ceremony like it only works with they underlying JavaScript implementation.

Reply:
In the end turns out we need to cast even for `styled-components`. They expect `CSSObject` there, which extends `CSS.Properties`, but it still complains if the object passed is typed as `CSS.Properties`. 

Here is a minimal repro: https://codesandbox.io/s/elastic-herschel-1w4p7?file=/src/App.tsx

Reply:
I reverted the styled components usage, until we solve this issue, I don't see a benefit for adding the examples at this moment

Reply:
Thanks for the reproduction, I was curious to see how emotion and sc interoperability was like https://codesandbox.io/s/amazing-snowflake-27h15?file=/src/App.tsx.